app/controllers/groups_controller.php

```diff
@@ -874,8 +908,36 @@ function _addGroupByImport($lines, $courseId, $update, $identifier, $render=true
      * @access public
      * @return void
      */
-    function export($courseId)
+    function export($courseId, $exportTo = 'file')
     {
+        $this->set('exportTo', $exportTo);
+
+        if ($exportTo == 'canvas') {
+
+            App::import('Component', 'CanvasCourse');
+
+            $this->data['Group']['syncType'] = 'export';
+            $this->data['Group']['updateCanvasGroups'] = true;
+            $this->data['Group']['selectAllGroups'] = true;
+
+            $this->syncCanvas($courseId);
+
+            $this->set('formUrl', Router::url(DS . $this->params['url']['url'], true));
+
+            // Fix breadcrumb
+            $this->breadcrumb->pop();
+            $this->breadcrumb->push(__('Export iPeer Groups to Canvas', true));
+            $this->set('breadcrumb', $this->breadcrumb);
+
+            // some error messages need to be updated to be relevant to this screen
+            if (!empty($this->errorMessages['groupExport']['noGroupsSelected'])){
```

Comment:
When there is no groups defined in iPeer and user choose to export to Cavans:
- no error message is displayed
- the same export selection screen is displayed
- all Canvas groups deleted